grammar.js

```diff
@@ -769,10 +774,26 @@ const rules = {
       field('name', $.identifier),
       opt($.type_parameters),
       $.parameters,
+      opt($.capability_list),
       opt(seq(':', opt($.attribute_modifier), field('return_type', $._type))),
       opt($.where_clause),
     ),
 
+  capability_list: $ => seq('[', opt(com($.capability)), ']'),
+
+  capability: $ =>
+    choice(
+      seq(
+        $.identifier,
+        opt($.type_parameters),
+      ),
+      $.scoped_identifier,
+      seq(
+        'ctx',
+        $.variable
```

Comment:
Great job covering this case too. Could you add a test for this one? I can't see a `[ctx $foo]` anywhere in your existing tests.

Reply:
This is tested in `test/cases/declarations/context-function-types.hack`